tests/test_list.py

```diff
@@ -0,0 +1,62 @@
+"""Test the list command."""
+# mypy: ignore-errors
+# flake8: noqa
+
+import argparse
+from typing import Tuple
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+
+import dfetch
+from dfetch.commands.list import List
+from dfetch.manifest.manifest import Manifest
+from dfetch.manifest.project import ProjectEntry
+
+
+def mock_manifest(projects):
+
+    project_mocks = []
+
+    for project in projects:
+        mock_project = Mock(spec=ProjectEntry)
+        mock_project.name = project["name"]
+        project_mocks += [mock_project]
+
+    return MagicMock(spec=Manifest, projects=project_mocks)
+
+
+@pytest.mark.parametrize(
+    "name, projects",
+    [
+        ("empty", []),
+        ("single_project", [{"name": "my_project"}]),
+        ("two_projects", [{"name": "first"}, {"name": "second"}]),
+    ],
+)
+def test_list(name, projects):
+
+    check = List()
+
+    with patch("dfetch.manifest.manifest.get_manifest") as mocked_get_manifest:
+        with patch("dfetch.project.make") as mocked_make:
+
+            mocked_get_manifest.return_value = (mock_manifest(projects), "/")
+
+            check(argparse.Namespace)
+
+            for project in projects:
+                mocked_make.return_value.remote.assert_called()
+
+def test_log_metadata():
```

Comment:
I agree with you @jgeudens that this test doesn't add a lot. The test above might be more useful. 
You could mock print_info_line and see that at least each project is called.
I think that it would be good to have at least a single unittest for list, so if we ever feel the need for more tests that we can extend something existing.

Reply:
You will need to:
- Line 46: replace `check` with `list`
- Line 42: replace `with patch("dfetch.commands.list.Metadata") as mocked_metadata` with `with patch("dfetch.list.get_logger") as mocked_logger`

Reply:
Looks good the unit test, this is a perfect starting setup if we make it more complex later